dashboard/src/modules/components/ComparisonComponent/index.jsx

```diff
@@ -28,16 +28,24 @@ const ComparisonComponent = () => {
   const navigate = useNavigate();
 
   const { datasets } = useSelector((state) => state.overview);
-  const { isCompareSwitchChecked, selectedResourceIds } = useSelector(
-    (state) => state.comparison
-  );
+  const {
+    isCompareSwitchChecked,
+    selectedResourceIds,
+    compareChartData,
+    chartData,
+  } = useSelector((state) => state.comparison);
   useEffect(() => {
-    if (datasets && datasets.length > 0) {
+    if (
+      datasets &&
+      datasets.length > 0 &&
+      !!!compareChartData.length &&
+      !!!chartData.length
+    ) {
```

Comment:
What does `!!!` accomplish?  `!!` converts a value to a boolean.  I presume that `!!!` inverts the value produced by `!!`, but `!` should produce the same result.